ext/DimensionalDataMakie.jl

```diff
@@ -464,6 +491,19 @@ function _categorical_or_dependent(A, ::Nothing)
     end
 end
 
+# Check for regular sampling on a lookup, throw an error if not.
+# Here, we assume 
+function _check_regular_or_categorical_sampling(l; axis = nothing)
+    if !(DD.isregular(l) || DD.iscategorical(l))
+        throw(ArgumentError("""
+        DimensionalDataMakie: The $(isnothing(axis) ? "" : "$axis-axis ")lookup is not regularly spaced, which is required for image-like plot types in Makie.
+        The lookup was:
+        $l
+
+        You can solve this by resampling your raster, or by using a more permissive plot type like `heatmap`, `surface`, `contour`, or `contourf`. 
```

Comment:
@rafaqz I just hit this edge case for raster's volume-like cases. what would be the way to get a regular-spaced raster? a better suggestion is needed here.  Maybe one-liner to resample a given raster `A`?

Reply:
Ah yeah a one liner would be good to add there.  I should also add an option to customize the error message so this can be used in volume plots...

Reply:
With `image` or `heatmap`? `heatmap` should wirk Maybe we can just warn for `image` and plot anyway?

Reply:
the issue I found was for something with `X, Y, Ti`, creating a volume, where the axis for Ti is usually irregular, hence failing afterwards to plot, I just want say, treat this as a regular thing as well, it makes sense for daily data, but I'm sure there should be others where we should not do that.

Reply:
Aha ok!  Yeah I guess it's a bit restrictive to error all the time.